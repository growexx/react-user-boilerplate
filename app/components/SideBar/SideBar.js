/* eslint-disable react/no-array-index-key */
/* eslint-disable react/no-access-state-in-setstate */
import React from 'react';
import { Layout, Menu } from 'antd';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { MenuItems } from './Constants';
// import StyledSideBarWrapper from './StyledSideBarWrapper';

const { Sider } = Layout;

const SideBar = props => (
  <Sider
    trigger={null}
    collapsible
    collapsed={props.collapsed}
    id="components-layout-demo-custom-trigger"
  >
    <div className="logo" />
    <Menu theme="dark" mode="inline" defaultSelectedKeys={['1']}>
      {MenuItems.map((menu, i) => (
        <>
          <Menu.Item key={i} icon={menu.icon}>
            <Link to={menu.to}>{menu.tabName}</Link>
          </Menu.Item>
        </>
      ))}
    </Menu>
  </Sider>
);

export default SideBar;

SideBar.propTypes = {
  collapsed: PropTypes.bool,
};
