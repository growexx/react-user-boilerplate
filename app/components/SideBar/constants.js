/*
 * Sidebar Constants
 * This file contains constants used in Sidebar component.
 */

import React from 'react';
import {
  UserOutlined,
  SmileOutlined,
  LoadingOutlined,
  ExportOutlined,
  NumberOutlined,
  FormOutlined,
  LockOutlined,
  PieChartOutlined,
  MessageOutlined,
  ShoppingOutlined,
  SearchOutlined,
} from '@ant-design/icons';
import {
  ROUTES,
  ROLE_BASED_SIDEBAR_MENU,
  ROLES,
  RESTRICTED_ROUTES,
} from 'containers/constants';

export const MenuItems = [
  {
    to: ROUTES.HOME,
    tabName: 'Home',
    icon: <SmileOutlined />,
  },
  {
    to: ROUTES.SAMPLE_FORM,
    tabName: 'Redux-Saga Form',
    icon: <FormOutlined />,
  },
  {
    to: ROUTES.LOADER,
    tabName: 'Loader Demo',
    icon: <LoadingOutlined />,
  },
  {
    to: ROUTES.EXPORT_DATA,
    tabName: 'Export Data To CSV',
    icon: <ExportOutlined />,
  },
  {
    to: ROUTES.NUMERAL_CONVERTER,
    tabName: 'Number Conversion Demo',
    icon: <NumberOutlined />,
  },
  {
    to: ROUTES.USERS,
    tabName: 'User Management',
    icon: <UserOutlined />,
  },
  {
    to: ROUTES.GITHUB_SEARCH,
    tabName: 'Github Search',
    icon: <SearchOutlined />,
  },
  {
    to: ROUTES.TEST_ADMIN_PAGE,
    tabName: 'Admin Page',
    icon: <LockOutlined />,
  },
  {
    to: ROUTES.CHARTS,
    tabName: 'Charts',
    icon: <PieChartOutlined />,
  },
  {
    to: ROUTES.MULTI_TAB_SUPPORT,
    tabName: 'Multi Tab Communication',
    icon: <MessageOutlined />,
  },
  {
    to: ROUTES.PRODUCTS,
    tabName: 'Products',
    icon: <ShoppingOutlined />,
  },
];

/**
 * Filters Sidebar menu based on role
 * @param {string} role
 */
export const GET_FILTERED_MENU_ITEM = role =>
  MenuItems.filter(item =>
    // Check if route is private then role has access to it
    RESTRICTED_ROUTES.includes(item.to)
      ? ROLE_BASED_SIDEBAR_MENU[role || ROLES.USER].includes(item.to)
      : true,
  );
