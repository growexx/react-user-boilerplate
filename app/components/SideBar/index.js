/**
 * SideBar/index.js
 *
 * This is the SideBar Component File.
 */
import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { Layout, Menu } from 'antd';
import PropTypes from 'prop-types';
import { ROUTES } from 'containers/constants';
import { showLogoInSideBar } from 'components/constants';
import GrowExxTriangleLogo from '../../images/Growexx-Triangle-White.png';
import GrowExxLogo from '../../images/GrowExx_Group_Logo.png';
import { GET_FILTERED_MENU_ITEM } from './constants';

const { Sider } = Layout;

const SideBar = props => {
  const Logo = !props.collapsed ? (
    <img src={GrowExxLogo} alt="logo" />
  ) : (
    <img src={GrowExxTriangleLogo} alt="logo" />
  );
  return (
    <Sider
      trigger={null}
      collapsible
      collapsed={props.collapsed}
      id="components-layout-demo-custom-trigger"
    >
      {showLogoInSideBar(props.layoutVariant) ? (
        <div className="logo">
          <Link to={ROUTES.HOME}>{Logo}</Link>
        </div>
      ) : null}
      <Menu
        theme="dark"
        mode="inline"
        defaultSelectedKeys={[props.location.pathname]}
        selectedKeys={[props.location.pathname]}
      >
        {GET_FILTERED_MENU_ITEM(props.user && props.user.role).map(menu => (
          <Menu.Item key={menu.to} icon={menu.icon}>
            <Link to={menu.to}>{menu.tabName}</Link>
          </Menu.Item>
        ))}
      </Menu>
    </Sider>
  );
};

export default withRouter(SideBar);

SideBar.propTypes = {
  collapsed: PropTypes.bool,
  user: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
  location: PropTypes.object.isRequired,
  layoutVariant: PropTypes.number,
};
