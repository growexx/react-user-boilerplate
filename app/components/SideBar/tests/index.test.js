import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import history from 'utils/history';
import SideBar from '../index';
import configureStore from '../../../configureStore';
let store;
const props = {
  user: {
    role: 1,
  },
  collapsed: true,
};
const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <SideBar {...props} />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

describe('<SideBar />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
});
