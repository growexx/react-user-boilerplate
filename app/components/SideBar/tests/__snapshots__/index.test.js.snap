// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> should render and match the snapshot 1`] = `
<aside
  class="ant-layout-sider ant-layout-sider-dark ant-layout-sider-collapsed"
  id="components-layout-demo-custom-trigger"
  style="flex: 0 0 80px; max-width: 80px; min-width: 80px; width: 80px;"
>
  <div
    class="ant-layout-sider-children"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          src="IMAGE_MOCK"
        />
      </a>
    </div>
    <ul
      class="ant-menu ant-menu-dark ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
      role="menu"
    >
      <li
        class="ant-menu-item ant-menu-item-selected"
        role="menuitem"
      >
        <span
          aria-label="smile"
          class="anticon anticon-smile ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="smile"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M288 421a48 48 0 1096 0 48 48 0 10-96 0zm352 0a48 48 0 1096 0 48 48 0 10-96 0zM512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm263 711c-34.2 34.2-74 61-118.3 79.8C611 874.2 562.3 884 512 884c-50.3 0-99-9.8-144.8-29.2A370.4 370.4 0 01248.9 775c-34.2-34.2-61-74-79.8-118.3C149.8 611 140 562.3 140 512s9.8-99 29.2-144.8A370.4 370.4 0 01249 248.9c34.2-34.2 74-61 118.3-79.8C413 149.8 461.7 140 512 140c50.3 0 99 9.8 144.8 29.2A370.4 370.4 0 01775.1 249c34.2 34.2 61 74 79.8 118.3C874.2 413 884 461.7 884 512s-9.8 99-29.2 144.8A368.89 368.89 0 01775 775zM664 533h-48.1c-4.2 0-7.8 3.2-8.1 7.4C604 589.9 562.5 629 512 629s-92.1-39.1-95.8-88.6c-.3-4.2-3.9-7.4-8.1-7.4H360a8 8 0 00-8 8.4c4.4 84.3 74.5 151.6 160 151.6s155.6-67.3 160-151.6a8 8 0 00-8-8.4z"
            />
          </svg>
        </span>
        <span>
          <a
            href="/"
          >
            Home
          </a>
        </span>
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
      >
        <span
          aria-label="form"
          class="anticon anticon-form ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="form"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M904 512h-56c-4.4 0-8 3.6-8 8v320H184V184h320c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V520c0-4.4-3.6-8-8-8z"
            />
            <path
              d="M355.9 534.9L354 653.8c-.1 8.9 7.1 16.2 16 16.2h.4l118-2.9c2-.1 4-.9 5.4-2.3l415.9-415c3.1-3.1 3.1-8.2 0-11.3L785.4 114.3c-1.6-1.6-3.6-2.3-5.7-2.3s-4.1.8-5.7 2.3l-415.8 415a8.3 8.3 0 00-2.3 5.6zm63.5 23.6L779.7 199l45.2 45.1-360.5 359.7-45.7 1.1.7-46.4z"
            />
          </svg>
        </span>
        <span>
          <a
            href="/sample-form"
          >
            Redux-Saga Form
          </a>
        </span>
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
      >
        <span
          aria-label="loading"
          class="anticon anticon-loading anticon-spin ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="loading"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
            />
          </svg>
        </span>
        <span>
          <a
            href="/loader"
          >
            Loader Demo
          </a>
        </span>
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
      >
        <span
          aria-label="export"
          class="anticon anticon-export ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="export"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M888.3 757.4h-53.8c-4.2 0-7.7 3.5-7.7 7.7v61.8H197.1V197.1h629.8v61.8c0 4.2 3.5 7.7 7.7 7.7h53.8c4.2 0 7.7-3.4 7.7-7.7V158.7c0-17-13.7-30.7-30.7-30.7H158.7c-17 0-30.7 13.7-30.7 30.7v706.6c0 17 13.7 30.7 30.7 30.7h706.6c17 0 30.7-13.7 30.7-30.7V765.1c0-4.3-3.5-7.7-7.7-7.7zm18.6-251.7L765 393.7c-5.3-4.2-13-.4-13 6.3v76H438c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 000-12.6z"
            />
          </svg>
        </span>
        <span>
          <a
            href="/export-data"
          >
            Export Data To CSV
          </a>
        </span>
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
      >
        <span
          aria-label="number"
          class="anticon anticon-number ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="number"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M872 394c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8H708V152c0-4.4-3.6-8-8-8h-64c-4.4 0-8 3.6-8 8v166H400V152c0-4.4-3.6-8-8-8h-64c-4.4 0-8 3.6-8 8v166H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h168v236H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h168v166c0 4.4 3.6 8 8 8h64c4.4 0 8-3.6 8-8V706h228v166c0 4.4 3.6 8 8 8h64c4.4 0 8-3.6 8-8V706h164c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8H708V394h164zM628 630H400V394h228v236z"
            />
          </svg>
        </span>
        <span>
          <a
            href="/number-converter-demo"
          >
            Number Conversion Demo
          </a>
        </span>
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
      >
        <span
          aria-label="user"
          class="anticon anticon-user ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="user"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </span>
        <span>
          <a
            href="/users"
          >
            User Management
          </a>
        </span>
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
      >
        <span
          aria-label="search"
          class="anticon anticon-search ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="search"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
        </span>
        <span>
          <a
            href="/github-search"
          >
            Github Search
          </a>
        </span>
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
      >
        <span
          aria-label="pie-chart"
          class="anticon anticon-pie-chart ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="pie-chart"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M864 518H506V160c0-4.4-3.6-8-8-8h-26a398.46 398.46 0 00-282.8 117.1 398.19 398.19 0 00-85.7 127.1A397.61 397.61 0 0072 552a398.46 398.46 0 00117.1 282.8c36.7 36.7 79.5 65.6 127.1 85.7A397.61 397.61 0 00472 952a398.46 398.46 0 00282.8-117.1c36.7-36.7 65.6-79.5 85.7-127.1A397.61 397.61 0 00872 552v-26c0-4.4-3.6-8-8-8zM705.7 787.8A331.59 331.59 0 01470.4 884c-88.1-.4-170.9-34.9-233.2-97.2C174.5 724.1 140 640.7 140 552c0-88.7 34.5-172.1 97.2-234.8 54.6-54.6 124.9-87.9 200.8-95.5V586h364.3c-7.7 76.3-41.3 147-96.6 201.8zM952 462.4l-2.6-28.2c-8.5-92.1-49.4-179-115.2-244.6A399.4 399.4 0 00589 74.6L560.7 72c-4.7-.4-8.7 3.2-8.7 7.9V464c0 4.4 3.6 8 8 8l384-1c4.7 0 8.4-4 8-8.6zm-332.2-58.2V147.6a332.24 332.24 0 01166.4 89.8c45.7 45.6 77 103.6 90 166.1l-256.4.7z"
            />
          </svg>
        </span>
        <span>
          <a
            href="/charts"
          >
            Charts
          </a>
        </span>
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
      >
        <span
          aria-label="message"
          class="anticon anticon-message ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="message"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M464 512a48 48 0 1096 0 48 48 0 10-96 0zm200 0a48 48 0 1096 0 48 48 0 10-96 0zm-400 0a48 48 0 1096 0 48 48 0 10-96 0zm661.2-173.6c-22.6-53.7-55-101.9-96.3-143.3a444.35 444.35 0 00-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6.3-119.3 12.3-174.5 35.9a445.35 445.35 0 00-142 96.5c-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9A449.4 449.4 0 00112 714v152a46 46 0 0046 46h152.1A449.4 449.4 0 00510 960h2.1c59.9 0 118-11.6 172.7-34.3a444.48 444.48 0 00142.8-95.2c41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5.3-60.9-11.5-120-34.8-175.6zm-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-.6 99.6-39.7 192.9-110.1 262.7z"
            />
          </svg>
        </span>
        <span>
          <a
            href="/multi-tab-support"
          >
            Multi Tab Communication
          </a>
        </span>
      </li>
      <li
        class="ant-menu-item"
        role="menuitem"
      >
        <span
          aria-label="shopping"
          class="anticon anticon-shopping ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="shopping"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M832 312H696v-16c0-101.6-82.4-184-184-184s-184 82.4-184 184v16H192c-17.7 0-32 14.3-32 32v536c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V344c0-17.7-14.3-32-32-32zm-432-16c0-61.9 50.1-112 112-112s112 50.1 112 112v16H400v-16zm392 544H232V384h96v88c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-88h224v88c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-88h96v456z"
            />
          </svg>
        </span>
        <span>
          <a
            href="/products"
          >
            Products
          </a>
        </span>
      </li>
    </ul>
  </div>
</aside>
`;
